invenio/b2share/modules/b2deposit/b2share_marc_handler.py

```diff
@@ -157,6 +158,12 @@ def add_file_info(rec, form, email, sub_id, recid):
     else:
         fft_status = 'firerole: allow email "{0}"\ndeny all'.format(
             email)
+
+    if 'embargo_till' in form:
+        embargodate = parser.parse(form['embargo_till'])
+        embargodate = datetime.strftime(embargodate, '%Y-%m-%d')
+        fft_status = 'firerole: deny until "%s"\nallow any\n' % embargodate
```

Comment:
As the first rule is `deny until ...` this will also deny the owner to access the record, and it's not what we want. I suppose the first rule must be `allow email ...` and then the deny rule.